--- origsrc/balsa-2.3.16/libbalsa/imap/imap-handle.c	2007-05-05 12:37:21.000000000 -0500
+++ src/balsa-2.3.16/libbalsa/imap/imap-handle.c	2007-05-31 15:41:02.276936200 -0500
@@ -40,6 +40,12 @@
 #include <openssl/err.h>
 #endif
 
+#ifdef __CYGWIN__
+#undef HAVE_GETADDRINFO
+#undef HAVE_GETNAMEINFO
+#include <getaddrinfo.h>
+#endif
+
 #include "libimap-marshal.h"
 #include "imap-auth.h"
 #include "imap-handle.h"
--- origsrc/balsa-2.3.16/src/print-gtk.c	2007-02-13 12:05:39.000000000 -0600
+++ src/balsa-2.3.16/src/print-gtk.c	2007-05-31 15:31:47.886486200 -0500
@@ -377,7 +377,7 @@
      */
     gchar *e = _("default:mm");
   
-#ifdef HAVE_LANGINFO
+#if defined(HAVE_LANGINFO) && defined(_NL_MEASUREMENT_MEASUREMENT)
     gchar *imperial = NULL;
   
     imperial = nl_langinfo(_NL_MEASUREMENT_MEASUREMENT);
